mux.js

```diff
@@ -0,0 +1,248 @@
+var protocol = require('hypercore-protocol')
+var readify = require('./ready')
+var inherits = require('inherits')
+var events = require('events')
+var debug = require('debug')('multifeed')
+var hypercore = require('hypercore')
+var xtend = require('xtend')
+
+// constants
+var MULTIFEED = 'MULTIFEED'
+var PROTOCOL_VERSION = '2.0.0'
+// extensions
+var MANIFEST = 'MANIFEST'
+var REQUEST_FEEDS = 'REQUEST_FEEDS'
+/*
+var ANNOUNCE_FEED = 'ANNOUNCE_FEED'
+var REQUEST_FEED_SIGNATURE = 'REQUEST_FEED_SIGNATURE'
+var FEED_SIGNATURE = 'FEED_SIGNATURE'
+*/
+
+var SupportedExtensions = [
+  MANIFEST,
+  REQUEST_FEEDS
+  //ANNOUNCE_FEED
+  //REQUEST_MANIFEST,
+  //REQUEST_FEED_SIGNATURE,
+  //FEED_SIGNATURE,
+]
+
+// `key` - protocol encryption key
+// `opts`- hypercore-protocol opts
+function Multiplexer (key, opts) {
+  if (!(this instanceof Multiplexer)) return new Multiplexer(key, opts)
+  debug('[REPLICATION] New mux initialized', key.toString('hex'), opts)
+  var self = this
+  self._opts = opts = opts || {}
+  self.extensions = opts.extensions = SupportedExtensions || opts.extensions
```

Comment:
```suggestion
  self.extensions = opts.extensions || SupportedExtensions
```

Reply:
Did you mean to have `opts.extensions` be used when passed and fallback to Supported, or actually vice versa?

Reply:
Oops resolved conversation by accident. I approve of the change. The original intention was to keep self.extensions and opts.extensions in sync thats why there's two lefthand assignments

Reply:
Sigh. Can't find an 'undo resolve' button..

Reply:
Yes verifying `opts.extensions` should be used when passed and `SupportedExtensions` is the fallback.